plugins/inputs/statsd/datadog_test.go

```diff
@@ -0,0 +1,353 @@
+package statsd
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventGather(t *testing.T) {
+	acc := &testutil.Accumulator{}
+
+	now := time.Now()
+	s := NewTestStatsd()
+	err := s.parseEventMessage(now, "_e{10,9}:test title|test text", "default-hostname")
+	require.Nil(t, err)
+	err = s.parseEventMessage(now.Add(1), "_e{10,24}:test title|test\\line1\\nline2\\nline3", "default-hostname")
+	require.Nil(t, err)
+	err = s.parseEventMessage(now.Add(2), "_e{10,9}:test title|test text|d:21", "default-hostname")
```

Comment:
This, and a few other places in this file, would be a great place to use a table test.  Please also use require.Equal instead of assert.Equal more or less everywhere.

Reply:
The reason I didn't use a table test was to make it more similar to their tests.  I wanted to ensure compatibility so I copied their tests over then converted them to our setup.

Reply:
I could switch, them over now though.

Reply:
I would switch them to the format we prefer, I'm don't anticipate that we will want to merge the code frequently.